pkg/client/record/events_cache.go

```diff
@@ -49,23 +54,35 @@ func AddOrUpdateEvent(newEvent *api.Event) History {
 	key := getEventKey(newEvent)
 	previousEvents.Lock()
 	defer previousEvents.Unlock()
-	previousEvents.table[key] =
+	previousEvents.cache.Add(
+		key,
 		History{
 			Count:           newEvent.Count,
 			FirstTimestamp:  newEvent.FirstTimestamp,
 			Name:            newEvent.Name,
 			ResourceVersion: newEvent.ResourceVersion,
-		}
-	return previousEvents.table[key]
+		})
+	return getEvent(key)
 }
 
 // GetEvent returns the entry corresponding to the given event, if one exists, otherwise a History object
-// with a count of 1 is returned.
+// with a count of 0 is returned.
 func GetEvent(event *api.Event) History {
```

Comment:
I might be missing something, but both ```getEvent``` and ```addOrUpdateEvent``` should be threadsafe because access to the cache is protected with a RWMutex, right?